src/Integration.Vsix/ErrorList/SonarErrorListDataSource.cs

```diff
@@ -66,30 +60,27 @@ internal SonarErrorListDataSource(ITableManagerProvider tableManagerProvider)
         public string SourceTypeIdentifier => StandardTableDataSources.ErrorTableDataSource;
 
         // Note: Error List is the only expected subscriber
-        public IDisposable Subscribe(ITableDataSink sink) => new SinkManager(this, sink);
-
-        #endregion
-
-        #region ISinkManagerRegister
-
-        public void AddSinkManager(ISinkManager manager)
+        public IDisposable Subscribe(ITableDataSink sink)
         {
-            lock (managers)
+            lock(sinks)
             {
-                managers.Add(manager);
+                sinks.Add(sink);
 
-                foreach (var factory in factories)
+                foreach(var factory in factories)
                 {
-                    manager.AddFactory(factory);
+                    sink.AddFactory(factory);
                 }
+
+                return new ExecuteOnDispose(() => Unsubscribe(sink));
```

Comment:
We need to hand VS a disposable object that it will dispose when the sink is no longer available.
We just need to remove our reference to the sink when this happens. That is essentially the only reason the `SinkManager` class was required.
I've replaced it with a general purpose disposable class that can be passed a delegate.

Reply:
That's a neat pattern, I like it!